controller/tap/server.go

```diff
@@ -275,6 +350,103 @@ func parseMethod(method string) *common.HttpMethod {
 	}
 }
 
+func makeByResourceMatch(match *public.TapByResourceRequest_Match) (*proxy.ObserveRequest_Match, error) {
+	// TODO: for now assume it's always a single, flat `All` match list
+	seq := match.GetAll()
+	if seq == nil {
+		return nil, errors.New("no match specified")
```

Comment:
how are these errors manifested? We should be returning errors that gRPC can turn into semantic status codes.  For instance, in this case I'd expect us to return [`Unimplemented`](https://godoc.org/google.golang.org/grpc/codes).

Reply:
Good point. I've modified all return sites in `TapByResource` to return gRPC errors. If functions lower in the stack return errors that need propagation, we now first check if it's a gRPC error. If it is, we propagate, if not, we wrap it in a gRPC error.